range_queries/sparse_table.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file sparse_table.cpp
+ * @brief Implementation of [Sparse Table](https://en.wikipedia.org/wiki/Range_minimum_query) data structure
+ *
+ * @details
+ * Sparse Table is a data structure, that allows answering range queries.
+ * It can answer most range queries in O(logn), but its true power is answering range minimum queries
+ * or equivalent range maximum queries). For those queries it can compute the answer in O(1) time.
+ *
+ * * Running Time Complexity \n
+ * * Build : O(NlogN) \n
+ * * Range Query : O(1) \n
+*/
+
+#include <vector>
+#include <cassert>
+#include <iostream>
+#include <algorithm>
+
+/**
+ * @namespace range_queries
+ * @brief Range Queries algorithms
+ */
+namespace range_queries {
+/**
+ * This function precomputes intial log table for further use.
+ * @param n value of the size of the input array
+ * @return corresponding vector of the log table
+ */
+std::vector<int> computeLogs(int n) {
+    std::vector<int> logs(n+5);
+    logs[1] = 0;
+    for (int i = 2 ; i < n ; i++) {
+        logs[i] = logs[i/2] + 1;
```

Comment:
In the case of `int` division, it considers `floor` by default. No need handle it explicitly.

Reply:
Note that this was oe of the explicit features added in C99. Before C99, the integer division was compiler specific. 

In this particular case, since we are setting the C standard to C99 in the CMake, this should compile and run fine. But since it is a division by 2, a no-hassle and fast division would be to shift-right. 
`logs[i] = logs[i >> 1] + 1;`